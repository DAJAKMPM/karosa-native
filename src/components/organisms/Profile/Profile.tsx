/**
 *
 * Profile
 * @format
 *
 */

import React, { FC } from "react";

import type { PropsType } from "./types";
import VariatonOne from "./VariationOne";

const Profile: FC<PropsType> = (props) => {
  const {
    variation,
    coverPhoto,
    avatarPhoto,
    searchProps,
    onBack,
    onMore,
    onChat,
    onAdd,
    shopName,
    address,
    isActive,
    rating,
    followers,
    chatPerf,
  } = props;

  switch (variation) {
    case 1:
      return (
        <VariatonOne
          coverPhoto={coverPhoto}
          avatarPhoto={avatarPhoto}
          searchProps={searchProps}
          onBack={onBack}
          onMore={onMore}
          onChat={onChat}
          onAdd={onAdd}
          shopName={shopName}
          address={address}
          isActive={isActive}
          rating={rating}
          followers={followers}
          chatPerf={chatPerf}
        />
      );

    default:
      return (
        <VariatonOne
          coverPhoto={coverPhoto}
          avatarPhoto={avatarPhoto}
          searchProps={searchProps}
          onBack={onBack}
          onMore={onMore}
          onChat={onChat}
          onAdd={onAdd}
          shopName={shopName}
          address={address}
          isActive={isActive}
          rating={rating}
          followers={followers}
          chatPerf={chatPerf}
        />
      );
  }
};

export default Profile;
