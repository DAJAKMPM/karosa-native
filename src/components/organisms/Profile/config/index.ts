/**
 *
 * @format
 *
 */

export const BLUR_RADIUS = 1;

export const ADDRESS_LINE_NO = 2;

export const ICON = {
  BUTTON: 16,
  ACTIVITY: 8,
};
