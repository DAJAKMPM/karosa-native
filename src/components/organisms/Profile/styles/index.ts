/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { DIMENS, SPACING, theme } from "@app/styles";

const ProfileStyles = StyleSheet.create({
  container: {
    position: "relative",
  },
  flexRow: {
    flexDirection: "row",
  },
  profileContainer: {
    backgroundColor: theme.colors.white,
    paddingHorizontal: 22,
  },
  spacer: {
    marginVertical: 32,
  },
  nameContainer: {
    maxWidth: 320,
    marginBottom: 10,
  },
  iconContainer: {
    flexDirection: "row",
    alignItems: "center",
    marginLeft: -1,
  },
  coverPhoto: {
    height: DIMENS.screenHeight * 0.22,
  },
  headingIcon: {
    marginRight: 14,
  },
  headerContainer: { backgroundColor: theme.colors.transparent },
  avatarContainer: {
    position: "absolute",
    top: DIMENS.screenHeight * 0.14,
    left: DIMENS.screenWidth * 0.038,
    zIndex: 1,
  },
  statusBarSpacer: {
    margin: SPACING.xs - 1,
  },
  avatarPhoto: {
    width: DIMENS.screenHeight * 0.15,
    height: DIMENS.screenHeight * 0.15,
    borderRadius: (DIMENS.screenHeight * 0.15) / 2,
    borderColor: theme.colors.white,
    borderWidth: 5,
  },
  profileInfoContainer: {
    height: DIMENS.screenHeight * 0.23,
    backgroundColor: theme.colors.white,
    zIndex: 0,
    borderBottomWidth: 1,
    borderBottomColor: theme.colors.light10,
  },
  chatIcon: {
    position: "absolute",
    left: 25,
  },
  txtChat: {
    ...theme.textLight,
    alignSelf: "center",
    position: "absolute",
    left: 50,
    color: theme.colors.dark20,
    fontWeight: "700",
  },
  btnChat: {
    position: "absolute",
    top: 13,
    right: 145,
    width: DIMENS.screenWidth * 0.3,
    height: DIMENS.screenHeight * 0.05,
    borderRadius: 3,
    backgroundColor: theme.colors.light10,
  },
  personIcon: {
    position: "absolute",
    left: DIMENS.screenWidth * 0.05,
  },
  settingsIcon: {
    position: "absolute",
    left: DIMENS.screenWidth * 0.05,
  },
  txtPerson: {
    ...theme.textLight,
    alignSelf: "center",
    position: "absolute",
    left: DIMENS.screenWidth * 0.12,
    fontWeight: "700",
  },
  btnPerson: {
    position: "absolute",
    right: DIMENS.screenWidth * 0.06,
    top: DIMENS.screenHeight * 0.01,
    width: DIMENS.screenWidth * 0.3,
    height: DIMENS.screenHeight * 0.05,
    borderRadius: 3,
  },
  shopNameContainer: {
    position: "relative",
    paddingRight: 18,
    top: DIMENS.screenHeight * 0.085,
    left: DIMENS.screenWidth * 0.045,
  },
  shopAddressSubContainer: {
    width: "70%",
    flexDirection: "row",
    marginBottom: 10,
  },
  txtShopName: {
    ...theme.textBold,
    fontWeight: "500",
  },
  shopAddrContainer: {
    position: "relative",
    top: DIMENS.screenHeight * 0.1,
    left: DIMENS.screenWidth * 0.045,
    flexDirection: "row",
  },
  txtShopAddr: {
    ...theme.textLight,
    fontWeight: "300",
  },
  activeContainer: {
    flexDirection: "row",
    marginLeft: SPACING.md,
    alignItems: "center",
  },
  profileIcon: {
    marginRight: 4,
  },
  txtIcon: {
    ...theme.textLightBold,
    marginLeft: SPACING.xs,
    fontWeight: "700",
  },
  bottomInfoContainer: {
    marginLeft: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    marginBottom: 16,
  },
  ratingContainer: {
    flexDirection: "column",
  },
  txtBottomValue: {
    ...theme.textLightBold,
    fontWeight: "700",
  },
  txtBottomLabel: {
    ...theme.textLight,
    fontWeight: "300",
  },
  followersContainer: {
    flexDirection: "column",
  },
  chatPerformanceContainer: {
    flexDirection: "column",
  },
  chatPerfSubContainer: {
    marginLeft: SPACING.xs,
  },
});

export default ProfileStyles;
