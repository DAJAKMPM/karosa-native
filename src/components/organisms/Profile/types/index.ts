/**
 *
 * @format
 *
 */

import { PropsType as SearchBarProps } from "@app/molecules/SearchBar/types";

export type PropsType = {
  variation: number;
  coverPhoto: string;
  avatarPhoto: string;
  searchProps?: SearchBarProps;
  onBack(): void;
  onMore?(): void;
  onChat?(): void;
  onFollow?(): void;
  onSettings?(): void;
  shopName: string;
  address: string;
  isActive: boolean;
  rating: string;
  followers: string;
  chatPerf: string;
};
