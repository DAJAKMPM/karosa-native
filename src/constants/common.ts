export const VARIATION = {
  ONE: 1,
  TWO: 2,
  THREE: 3,
  FOUR: 4,
};

export const INIT_PICKER_VALUE = -1;

export const REGEX = {
  name: /^[a-zA-Z .,]+$/,
  phoneNumber:
    /^[\\+]?([0-9][\s]?|[0-9]?)([(][0-9]{3}[)][\s]?|[0-9]{3}[-\s\\.]?)[0-9]{3}[-\s\\.]?[0-9]{4,6}$/,
};
