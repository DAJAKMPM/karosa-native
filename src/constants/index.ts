import LIMITS from "./limits";
import * as ENUM from "./enums";
import {
  NavigationType as NAVIGATION_TYPE,
  LoginHeaderProps as LOGIN_HEADER_PROPS,
} from "./types";

export { LIMITS, ENUM, NAVIGATION_TYPE, LOGIN_HEADER_PROPS };
