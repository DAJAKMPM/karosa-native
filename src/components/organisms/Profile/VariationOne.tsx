/**
 *
 * VariationOne
 * @format
 *
 */

import React, { FC, Fragment } from "react";
import { ImageBackground, StatusBar, View } from "react-native";
import Image from "@app/atoms/Image";
import Button from "@app/atoms/Button";
import Header from "@app/molecules/Header";
import Searchbar from "@app/molecules/SearchBar";
import Icon from "@app/atoms/Icon";
import Text from "@app/atoms/Text";

import type { VariatonOne } from "./types";
import { BLUR_RADIUS, ICON } from "./config";
import ProfileStyles from "./styles";

const VariationOne: FC<VariatonOne> = (props) => {
  const {
    coverPhoto,
    avatarPhoto,
    searchProps,
    onBack,
    onMore,
    onChat,
    onAdd,
    shopName,
    address,
    isActive,
    rating,
    followers,
    chatPerf,
  } = props;

  return (
    <Fragment>
      <ImageBackground
        style={ProfileStyles.coverPhoto}
        blurRadius={BLUR_RADIUS}
        source={{ uri: coverPhoto }}>
        <StatusBar translucent backgroundColor="transparent" />
        <View style={ProfileStyles.statusBarSpacer} />
        <Header
          customStyle={ProfileStyles.headerContainer}
          barStyle="light-content"
          leftComponent={{
            icon: "arrow-back",
            color: "white",
            onPress: onBack,
          }}
          centerComponent={<Searchbar {...searchProps} />}
          rightComponent={{
            icon: "more-horiz",
            color: "white",
            onPress: onMore,
          }}
        />
      </ImageBackground>
      <View style={ProfileStyles.avatarContainer}>
        <Image
          customStyle={ProfileStyles.avatarPhoto}
          source={{ uri: avatarPhoto }}
        />
      </View>
      <View style={ProfileStyles.profileInfoContainer}>
        <Button
          icon={
            <Icon
              group="common"
              name="chat"
              width={ICON.BUTTON}
              height={ICON.BUTTON}
              extraStyle={ProfileStyles.chatIcon}
            />
          }
          onPress={onChat}
          titleStyle={ProfileStyles.txtChat}
          customStyle={ProfileStyles.btnChat}
          title={"Chat"}
        />
        <Button
          icon={
            <Icon
              group="products"
              name="add"
              width={ICON.BUTTON}
              height={ICON.BUTTON}
              extraStyle={ProfileStyles.personIcon}
            />
          }
          onPress={onAdd}
          titleStyle={ProfileStyles.txtPerson}
          customStyle={ProfileStyles.btnPerson}
          title={"Chat"}
        />
        <View style={ProfileStyles.shopNameContainer}>
          <Text text={shopName} customStyle={ProfileStyles.txtShopName} />
        </View>
        <View style={ProfileStyles.shopAddrContainer}>
          <Text text={address} customStyle={ProfileStyles.txtShopAddr} />
          <View style={ProfileStyles.activeContainer}>
            <Fragment>
              <Icon
                group="common"
                name={isActive ? "greenCircle" : "grayCircle"}
                height={ICON.ACTIVITY}
                width={ICON.ACTIVITY}
                extraStyle={ProfileStyles.activeIcon}
              />
              <Text
                text={isActive ? "active" : "inactive"}
                customStyle={ProfileStyles.txtIcon}
              />
            </Fragment>
          </View>
        </View>
        <View style={ProfileStyles.bottomInfoContainer}>
          <View style={ProfileStyles.ratingContainer}>
            <Text
              text={`${rating} / 5.0`}
              customStyle={ProfileStyles.txtBottomValue}
            />
            <Text
              text="Shop Rating"
              customStyle={ProfileStyles.txtBottomLabel}
            />
          </View>
          <View style={ProfileStyles.followersPerfContainer}>
            <Text text={followers} customStyle={ProfileStyles.txtBottomValue} />
            <Text text="Followers" customStyle={ProfileStyles.txtBottomLabel} />
          </View>
          <View style={ProfileStyles.followersPerfContainer}>
            <Text text={chatPerf} customStyle={ProfileStyles.txtBottomValue} />
            <Text
              text=" Chat Performance"
              customStyle={ProfileStyles.txtBottomLabel}
            />
          </View>
        </View>
      </View>
    </Fragment>
  );
};

export default VariationOne;
