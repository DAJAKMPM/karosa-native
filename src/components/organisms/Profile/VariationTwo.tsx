/**
 *
 * VariationTwo
 * @format
 *
 */

import React, { FC, Fragment } from "react";
import { ImageBackground, StatusBar, View } from "react-native";
import Image from "@app/atoms/Image";
import Button from "@app/atoms/Button";
import Header from "@app/molecules/Header";
import Icon from "@app/atoms/Icon";
import Text from "@app/atoms/Text";

import type { PropsType } from "./types";
import { BLUR_RADIUS, ICON } from "./config";
import ProfileStyles from "./styles";

const VariationTwo: FC<PropsType> = (props) => {
  const {
    coverPhoto,
    avatarPhoto,
    onBack,
    onChat,
    onSettings,
    shopName,
    address,
    isActive,
    rating,
    followers,
    chatPerf,
  } = props;

  return (
    <Fragment>
      <ImageBackground
        style={ProfileStyles.coverPhoto}
        blurRadius={BLUR_RADIUS}
        source={{ uri: coverPhoto }}>
        <StatusBar translucent backgroundColor="transparent" />
        <View style={ProfileStyles.statusBarSpacer} />
        <Header
          containerStyle={ProfileStyles.headerContainer}
          barStyle="light-content"
          leftComponent={{
            icon: "arrow-back",
            color: "white",
            onPress: onBack,
          }}
          rightComponent={{
            icon: "chat",
            color: "white",
            onPress: onChat,
          }}
        />
      </ImageBackground>
      <View style={ProfileStyles.avatarContainer}>
        <Image
          imageStyle={ProfileStyles.avatarPhoto}
          source={{ uri: avatarPhoto }}
        />
      </View>
      <View style={ProfileStyles.profileInfoContainer}>
        <Button
          icon={
            <Icon
              group="common"
              name="settingsWhite"
              width={ICON.BUTTON}
              height={ICON.BUTTON}
              extraStyle={ProfileStyles.settingsIcon}
            />
          }
          onPress={onSettings}
          titleStyle={ProfileStyles.txtPerson}
          buttonStyle={ProfileStyles.btnPerson}
          title={"Settings"}
        />
        <View style={ProfileStyles.shopNameContainer}>
          <Text text={shopName} textStyle={ProfileStyles.txtShopName} />
        </View>
        <View style={ProfileStyles.shopAddrContainer}>
          <Text text={address} textStyle={ProfileStyles.txtShopAddr} />
          <View style={ProfileStyles.activeContainer}>
            <Fragment>
              <Icon
                group="common"
                name={isActive ? "greenCircle" : "grayCircle"}
                height={ICON.ACTIVITY}
                width={ICON.ACTIVITY}
                extraStyle={ProfileStyles.activeIcon}
              />
              <Text
                text={isActive ? "Active" : "Inactive"}
                textStyle={ProfileStyles.txtIcon}
              />
            </Fragment>
          </View>
        </View>
        <View style={ProfileStyles.bottomInfoContainer}>
          <View style={ProfileStyles.ratingContainer}>
            <Text
              text={`${rating} / 5.0`}
              textStyle={ProfileStyles.txtBottomValue}
            />
            <Text text="Shop Rating" textStyle={ProfileStyles.txtBottomLabel} />
          </View>
          <View style={ProfileStyles.followersContainer}>
            <Text text={followers} textStyle={ProfileStyles.txtBottomValue} />
            <Text text="Followers" textStyle={ProfileStyles.txtBottomLabel} />
          </View>
          <View style={ProfileStyles.chatPerformanceContainer}>
            <View style={ProfileStyles.chatPerfSubContainer}>
              <Text text={chatPerf} textStyle={ProfileStyles.txtBottomValue} />
            </View>
            <Text
              text=" Chat Performance"
              textStyle={ProfileStyles.txtBottomLabel}
            />
          </View>
        </View>
      </View>
    </Fragment>
  );
};

export default VariationTwo;
