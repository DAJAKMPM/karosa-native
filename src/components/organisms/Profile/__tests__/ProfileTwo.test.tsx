import React from "react";
import { noop } from "lodash";
import { renderWithWrapper } from "@app/ci/testHelper";
import { COMMON } from "@app/constants";
import Profile from "../Profile";

const FAKE_URI = "https://i.imgur.com/0y8Ftya.jpg";

describe("Profile variation two Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Profile
        address="test"
        avatarPhoto={FAKE_URI}
        chatPerf="2"
        coverPhoto={FAKE_URI}
        followers="2"
        isActive
        onBack={noop}
        rating="2"
        shopName="test"
        variation={COMMON.VARIATION.TWO}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
