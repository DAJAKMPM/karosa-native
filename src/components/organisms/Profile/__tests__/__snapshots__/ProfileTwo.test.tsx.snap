// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile variation two Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": 293.48,
        }
      }
    >
      <Image
        blurRadius={1}
        source={
          Object {
            "uri": "https://i.imgur.com/0y8Ftya.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 293.48,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "margin": 3,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomColor": "#f2f2f2",
            "borderBottomWidth": 0,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
        testID="headerContainer"
      >
        <RNCSafeAreaView
          edges={
            Array [
              "left",
              "top",
              "right",
            ]
          }
          style={
            Object {
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "overflow": "hidden",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 3,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View>
                  <SvgMock
                    height={24}
                    style={
                      Object {
                        "marginRight": 14,
                      }
                    }
                    width={24}
                  />
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View>
                  <SvgMock
                    height={24}
                    width={24}
                  />
                </View>
              </View>
            </View>
          </View>
        </RNCSafeAreaView>
      </View>
    </View>
    <View
      style={
        Object {
          "left": 28.5,
          "position": "absolute",
          "top": 186.76000000000002,
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          onLoad={[Function]}
          source={
            Object {
              "uri": "https://i.imgur.com/0y8Ftya.jpg",
            }
          }
          style={
            Object {
              "borderColor": "#fff",
              "borderRadius": 100.05,
              "borderWidth": 5,
              "bottom": 0,
              "height": 200.1,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 200.1,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "borderColor": "#fff",
                "borderRadius": 100.05,
                "borderWidth": 5,
                "height": 200.1,
                "justifyContent": "center",
                "width": 200.1,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#1C1C1C"
                hidesWhenStopped={true}
                size="small"
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 50,
                "width": 50,
              },
              Object {
                "borderColor": "#fff",
                "borderRadius": 100.05,
                "borderWidth": 5,
                "height": 200.1,
                "width": 200.1,
              },
            ]
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "paddingHorizontal": 22,
        }
      }
    >
      <View
        style={
          Object {
            "marginVertical": 32,
          }
        }
      />
      <View
        style={
          Object {
            "marginBottom": 10,
            "maxWidth": 320,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1C1C1C",
              "fontFamily": "SFProText-Bold",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 19,
            }
          }
        >
          test
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
            "width": "70%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": -1,
            }
          }
        >
          <View>
            <SvgMock
              height={13}
              style={
                Object {
                  "marginRight": 4,
                }
              }
              width={13}
            />
          </View>
          <Text
            numberOfLines={2}
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "SFProText-Regular",
                "fontSize": 12,
                "fontWeight": "300",
                "lineHeight": 14.06,
              }
            }
          >
            test
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 12,
            }
          }
        >
          <View>
            <SvgMock
              height={8}
              style={
                Object {
                  "marginRight": 4,
                }
              }
              width={8}
            />
          </View>
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "SFProText-Bold",
                "fontSize": 12,
                "fontWeight": "700",
                "lineHeight": 14.06,
                "marginLeft": 4,
              }
            }
          >
            Active
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 16,
            "marginLeft": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "SFProText-Bold",
                "fontSize": 12,
                "fontWeight": "700",
                "lineHeight": 14.06,
              }
            }
          >
            2 / 5.0
          </Text>
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "SFProText-Regular",
                "fontSize": 12,
                "fontWeight": "300",
                "lineHeight": 14.06,
              }
            }
          >
            Shop Rating
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "SFProText-Bold",
                "fontSize": 12,
                "fontWeight": "700",
                "lineHeight": 14.06,
              }
            }
          >
            2
          </Text>
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "SFProText-Regular",
                "fontSize": 12,
                "fontWeight": "300",
                "lineHeight": 14.06,
              }
            }
          >
            Followers
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontFamily": "SFProText-Bold",
                  "fontSize": 12,
                  "fontWeight": "700",
                  "lineHeight": 14.06,
                }
              }
            >
              2
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "SFProText-Regular",
                "fontSize": 12,
                "fontWeight": "300",
                "lineHeight": 14.06,
              }
            }
          >
             Chat Performance
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
